dspace-spring-rest/src/main/java/org/dspace/app/rest/RelationshipRestController.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.UUID;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang3.StringUtils;
+import org.dspace.app.rest.converter.RelationshipConverter;
+import org.dspace.app.rest.link.HalLinkService;
+import org.dspace.app.rest.model.RelationshipRest;
+import org.dspace.app.rest.model.RelationshipRestWrapper;
+import org.dspace.app.rest.model.hateoas.RelationshipResourceWrapper;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.content.Item;
+import org.dspace.content.Relationship;
+import org.dspace.content.RelationshipType;
+import org.dspace.content.service.ItemService;
+import org.dspace.content.service.RelationshipService;
+import org.dspace.content.service.RelationshipTypeService;
+import org.dspace.core.Context;
+import org.dspace.util.UUIDUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This will be the entry point for the api/core/relationships endpoint with additional paths to it
+ */
+@RestController
+@RequestMapping("/api/core/relationships")
+public class RelationshipRestController {
```

Comment:
This has been included in https://jira.duraspace.org/browse/DS-4225

Reply:
@abollini : Do we have examples elsewhere of the "uniform behavior" that you are talking about?  I was looking around to see how this has been implemented on other endpoints (to see how hard this would be to refactor).  But, I was not able to find a good example.

Reply:
we should have no example for that because we typically don't need to have a separate controller. We use a front controller design patter here, where the actual logic is implemented by the REST Repository. 
Everything goes trough the RestResouceController, the only exception that I can remember are the bitstream controller and the opensearch controller that have very specific needs.